rero_ils/modules/notifications/dispatcher.py

```diff
@@ -19,127 +19,260 @@
+            #         f'Notification: {notification["pid"]} '
+            #         f'chanel: {patron["patron"]["communication_channel"]} '
+            #         f'type: {notification_type} '
+            #         f'loan: {loan["pid"]}'
+            #     )
+        return {'processed': count, 'send': send, 'not_send': not_send}
 
     @staticmethod
-    def _create_email(data, patron, library, recipients):
-        """."""
-        from flask_babelex import Locale
-
-        from .api import get_template_to_use
-        from ..loans.api import Loan
-
-        language = patron['patron']['communication_language']
-        # set the current language for translations in template
-        with current_app.test_request_context() as ctx:
-            ctx.babel_locale = Locale.parse(language)
-            loan = Loan.get_record_by_pid(data['loan']['pid'])
-            tpl_path = get_template_to_use(loan, data).rstrip('/')
-            template = f'{tpl_path}/{language}.txt'
-            # get the sender email from
-            # loan.pickup_location_pid.location.library.email
-            sender = library['email']
-            msg = TemplatedMessage(
-                template_body=template,
-                sender=sender,
-                recipients=recipients,
-                ctx=data['loan']
-            )
-            text = msg.body.split('\n')
-            # subject is the first line
-            msg.subject = text[0]
-            # body
-            msg.body = '\n'.join(text[1:])
-            return msg
+    def _create_email(recipients, sender, ctx_data, template):
+        """Create email message from template.
+
+        :param recipients: List of emails to send the message too.
+        :param sender: Sender email address.
+        :param ctx_data: Dictionair with informations used in template.
+        :param template: Template to use to create TemplatedMessage.
+        :returns: Message created.
+        """
+        msg = TemplatedMessage(
+            template_body=template,
+            sender=sender,
+            recipients=recipients,
+            ctx=ctx_data
+        )
+        text = msg.body.split('\n')
+        # subject is the first line
+        msg.subject = text[0]
+        # body
+        msg.body = '\n'.join(text[1:])
+        return msg
 
     @staticmethod
-    def send_mail_for_printing(data):
-        """Send the notification by email."""
-        patron = data['loan']['patron']
-        library = Location.get_record_by_pid(
-            data['loan']['pickup_location_pid']).get_library()
+    def send_mail_for_printing(ctx_data):
+        """Send the notification by email to the library.
+
+        :param patron: Patron used.
+        :param library: Library used for recipients email creation.
+        :param ctx_data: dDictionair with informations used in template.
+        :param notification_type: Notification type.
+        :param template: Template to use to create TemplatedMessage.
```

Comment:
params `patron`, `library`, `notification_type` and `template` are not present into the function signature.
typo into `dDictionair` word